app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,37 @@
+class SalesController < ApplicationController
+  before_action :get_vendor, only: [:index, :current_month, :new]
+
+  def get_vendor
+    @vendor = Vendor.find(params[:vendor_id])
+  end
+
+  def index
+    @sales = @vendor.sales.order(purchase_time: :desc)
+    @total_sales = Vendor.total_sales(@vendor.id).to_f / 100
+  end
+
+  def current_month
+    @current_month_sales = Sale.current_month_sales(@vendor.id).order(purchase_time: :desc)
+    @current_month_revenue = Sale.current_month_revenue(@vendor.id).to_f / 100
```

Comment:
Since these methods (and the `total_sales`) methods are applicable to a single vendor, it might make sense to make them instance methods on the specific vendor rather than class methods on the classes